gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java

```diff
@@ -423,6 +439,20 @@ private void registerRouteRegistrationHandler(Handler<Void> readyHandler) {
         readyHandler.handle(null);
     }
 
+    /**
+     * Update all registered router's pool size multiplier
+     *
+     * @param readyHandler - the ready handler
+     */
+    private void registerRouteMultiplierChangeHandler(Handler<Void> readyHandler) {
+        vertx.eventBus().consumer(Router.ROUTE_MULTIPLIER_ADRESS, (Handler<Message<String>>) event -> {
```

Comment:
Where are the messages to _Router.ROUTE_MULTIPLIER_ADRESS_ created? I only find consumers.

Reply:
I have a utility class in my project that repeatedly determines the number of nodes currently up and running


    public static int getRouteMultiplier(Vertx vertx){
        int numberOfClusterNodes = 1;
        ClusterManager clusterManager = ((VertxImpl)vertx).getClusterManager();
        if (clusterManager != null){
            numberOfClusterNodes = clusterManager.getNodes().size();
        }
        return Props.getNumberOfMainVerticleInstances() * numberOfClusterNodes;
    }
    
  
    Note that in my case the number of Verticles that create a Router is fixed and doesn't change at runtime.
    
    
It then publishes like this if things change

    private static void startRouteMultiplierPublishing(Vertx vertx){
        vertx.setPeriodic(10 * 1000, event -> {
            final EventBus eb = vertx.eventBus();
            int newRouteMultiplier = MainVerticle.getRouteMultiplier(vertx);
            if (previousRouteMultiplier != newRouteMultiplier) {
                eb.publish(Router.ROUTE_MULTIPLIER_ADRESS, String.valueOf(MainVerticle.getRouteMultiplier(vertx)));
                previousRouteMultiplier = newRouteMultiplier;
            }
        });
    }

Reply:
OK. I was looking for an API endpoint or a configuration resource. But this could be added "manually" in Crush/Houston if required.